android-components/components/browser/state/src/main/java/mozilla/components/browser/state/reducer/TranslationsStateReducer.kt

```diff
@@ -179,12 +178,12 @@ internal object TranslationsStateReducer {
         }
 
         is TranslationsAction.SetSupportedLanguagesAction ->
-            state.copyWithTranslationsState(action.tabId) {
-                it.copy(
+            state.copy(
+                translationEngine = state.translationEngine.copy(
                     supportedLanguages = action.supportedLanguages,
-                    translationError = null,
-                )
-            }
+                    engineError = null,
```

Comment:
I was trying to understand the implications of this logic a bit more and verify the correctness of whether or not we needed to clear `engineError` here. Couple of things I found was that our list of potential [TranslationError](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/translate/TranslationError.kt#17) is quite long. However, I think there is also a distinction between engine errors and the other errors.

The only errors that could be an `engineError` is https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt#163 and the newly introduced `CouldNotLoadLanguagesError` in this PR.

I think based on the sequence of events from my understanding of reading the code so far - (1) requestEngineSupport() and then (2) requestSupportedLanguages() in `InitAction` in https://github.com/mozilla-mobile/firefox-android/pull/5680/files#diff-a3c7bcd14d5c6f201fc9d9279918dc1b83ceba808076a42d0f3798ea000ffa62R49. There might not be a point of clearing this `engineError`. We would have an engine error to clear because requesting the engine support reported that we have an `UnknownEngineSupportError`  from `requestEngineSupport()`. We would then call `requestSupportedLanguages()`, but at that point, I don't know if there is a point if we don't have engine support.

Not sure if that made sense so let me know if this was confusing to follow.